.github/workflows/ci.yml

```diff
@@ -47,74 +48,83 @@ jobs:
         name: Set up Docker Buildx
         uses: docker/setup-buildx-action@v1
       -
-        name: Build Docker Image
-        uses: docker/build-push-action@v2
+        name: Login to Docker Hub
+        uses: docker/login-action@v1
         with:
-          file: "devops/dockerfiles/node/Dockerfile"
-          load: true
-          tags: mangatasolutions/mangata-node-fmt-check:${{ github.sha }}
-          target: env_builder
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Pull builder image
+        run: |
+          docker pull ${{ env.DOCKER_BUILDER_IMAGE }}
       -
         name: Check formatting
         run: |
-          docker run --entrypoint /root/.cargo/bin/cargo mangatasolutions/mangata-node-fmt-check:${{ github.sha }} fmt -- --check
+          ./docker-cargo.sh fetch || ./docker-cargo.sh fetch || ./docker-cargo.sh fetch || ./docker-cargo.sh fetch || ./docker-cargo.sh fetch
+          ./docker-cargo.sh fmt --all -- --check
 
-  clippy:
+  clippy-check:
     name: Clippy check
     runs-on: ubuntu-latest
-    if: "!contains(github.event.pull_request.labels.*.name, 'skip-build') && !${{ github.event.inputs.e2eBranch }}"
-
     steps:
-      - uses: actions/checkout@v2
-      - name: Install latest nightly
-        uses: actions-rs/toolchain@v1
-        with:
-            toolchain: nightly-2021-10-19
-            target: wasm32-unknown-unknown
-            override: true
-            components: clippy
-
-      - name: Run cargo check
-        uses: actions-rs/cargo@v1
+      -
+        name: Checkout
+        uses: actions/checkout@v2
+      -
+        name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      -
+        name: Login to Docker Hub
+        uses: docker/login-action@v1
         with:
-          command: clippy
-          args: -p pallet-xyk -p pallet-assets-info -p pallet-bridge
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Pull builder image
+        run: |
+          docker pull ${{ env.DOCKER_BUILDER_IMAGE }}
+      -
+        name: Run clippy
+        run: |
+          docker pull mangatasolutions/node-builder:0.1
+          ./docker-cargo.sh fetch || ./docker-cargo.sh fetch || ./docker-cargo.sh fetch || ./docker-cargo.sh fetch || ./docker-cargo.sh fetch
```

Comment:
why this? :O

Reply:
https://stackoverflow.com/questions/68283463/cargo-fails-to-download-dependencies-when-running-in-github-actions

was struggling with that for several days and couldn't figure out the solution. the cargo we have is even newer than mentioned in above post but problem still occasionally exists